tests/lib/hardware/hardware_base.py

```diff
@@ -28,9 +28,9 @@
 from typing import Dict, List
 import threading
 
+from tests.config import settings
```

Comment:
I would prefer to not import the settings now in every lib module. 
Instead we could just add a `keep_env` variable in the workspace, pass that in the constructor and then use the variable from there everywhere. The workspace is already available afaics

Reply:
Most modules/libraries are going to use settings of some form, so I don't really see the harm here. What is the advantage of moving it onto the workspace instance?

(Additionally this import already existed I was just re-ordering it alphabetically).

Reply:
I thought about cyrcular deps but given that the import is anyway there, it's fine for me